File: dart/src/n04_observer/editor.dart
import 'dart:io';

import 'event_manager.dart';
import 'event_type.dart';

class Editor {
    EventManager events = EventManager();

    void openFile(final File file) {
        this.events.notify(EventType.open, file);
    }

    void saveFile(final File file) {
        this.events.notify(EventType.save, file);
    }
}

File: java/src/n04_observer/Editor.java
package n04_observer;

import java.io.File;

public class Editor {
    public EventManager events;

    public Editor() {
        this.events = new EventManager();
    }

    public void openFile(File file) {
        this.events.notify(EventType.OPEN, file);
    }

    public void saveFile(File file) {
        this.events.notify(EventType.SAVE, file);
    }
}
